src/rgw/rgw_admin.cc

```diff
@@ -5504,6 +5508,86 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT_BUCKET_CHOWN) {
+    int ret = 0;
+    std::string tenant;
+    std::vector<rgw_bucket_dir_entry> objs;
+    map<string, bool> common_prefixes;
+    RGWObjectCtx obj_ctx(store);
+
+    //Get bucket info
+    RGWBucketInfo bucket_info;
+    RGWSysObjectCtx sys_ctx = store->svc.sysobj->init_obj_ctx();
+    map<string, bufferlist> attrs;
+
+    //split tenant/bucket
+    auto pos = bucket_name.find('/');
+    if (pos != boost::string_ref::npos) {
+      tenant = bucket_name.substr(0, pos);
+      bucket_name = bucket_name.substr(pos + 1);
+    }
+
+    ret = store->get_bucket_info(sys_ctx, tenant, bucket_name, bucket_info, NULL, &attrs);
+    if (ret < 0) {
+      cerr << " Bucket info failed: tenant: "<< tenant << "bucket_name: " << bucket_name << " " << ret << std::endl;
+      return ret;
+    }
+
+    //Get User info - for display name
+    RGWUserInfo user_info;
+    ret = rgw_get_user_info_by_uid(store, bucket_info.owner, user_info);
```

Comment:
This is because I was trying to avoid passing --uid in the command, because the object acls would have to just point to whatever the bucket's new owner is(by way of passing bucket_info.owner), assuming that the bucket link has already been run. Same with the --tenant option above.

Reply:
ok, i didn't realize this was fetching the bucket owner's user - following up on this below